iothub/device/src/AuthenticationWithTokenRefresh.cs

```diff
@@ -65,21 +88,26 @@ public abstract class AuthenticationWithTokenRefresh : IAuthenticationMethod, ID
         /// </summary>
         public bool IsExpiring => (ExpiresOn - DateTime.UtcNow).TotalSeconds <= _bufferSeconds;
 
-        // This internal property is used by the sdk to determine if the instance was created by the sdk,
-        // and thus, if it should be disposed by the sdk.
-        internal bool InstanceCreatedBySdk { get; set; }
+        // This internal property is used by the sdk to determine if the disposal should be handled by the sdk.
+        internal bool DisposalBySdk { get; }
 
         /// <summary>
         /// Gets a snapshot of the security token associated with the device. This call is thread-safe.
         /// </summary>
         public async Task<string> GetTokenAsync(string iotHub)
         {
+            if (_isDisposed)
```

Comment:
I made this fail-fast, since using a disposed authentication method might introduce some other issues down the line.